freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -358,14 +349,65 @@ private SyncStatusDetail internalSynchronizeStack(Stack stack, UmsUsersState ums
                 LOGGER.debug("Finished {}.", LogEvent.SYNC_CLOUD_IDENTITIES);
             }
 
+            if (fullSync && !warnings.isEmpty() && fmsToFreeipaBatchCallEnabled) {
+                UsersStateDifference usersStateDifferenceAfterSync = compareUmsAndFreeIpa(umsUsersState, fullSync, freeIpaClient);
+                if (usersStateDifferenceChanged(usersStateDifferenceBeforeSync, usersStateDifferenceAfterSync)) {
+                    LOGGER.info(String.format("Sync was partially successful for %s, thus we are trying it once again", stack.getResourceCrn()));
+                    warnings.clear();
```

Comment:
I know that we aren't really doing anything with the warnings yet, but I fear that we will lose information by clearing them here. We will return only those warnings from the retry. E.g., If the retry fails with some exception before adding a user, then the warnings won't indicate that the user wasn't added.

Reply:
sure, makes sense to keep warningsfrom original sync